app/src/components/market/common/market_scale/index.tsx

```diff
@@ -440,10 +472,28 @@ export const MarketScale: React.FC<Props> = (props: Props) => {
       setProfitLoss((amountSharesNumber || 0) * (1 - Number(scaleBall?.value) / 100) - (amountNumber || 0))
     } else {
       if (shortShares && collateral && !isDust(shortShares, collateral.decimals)) {
-        setShortPayout((shortSharesNumber || 0) * (1 - Number(scaleBall?.value) / 100))
+        const totalShortPriceNumber = Number(formatBigNumber(totalShortPrice, collateral.decimals, collateral.decimals))
+        const shortPayoutAmount = (shortSharesNumber || 0) * (1 - Number(scaleBall?.value) / 100)
+        const shortProfit = shortPayoutAmount - totalShortPriceNumber
+        setShortPayout(shortPayoutAmount)
+        setShortProfitAmount(shortProfit)
+        setShortProfitPercentage(
+          (shortPayoutAmount / totalShortPriceNumber - 1) * 100 < -100
+            ? -100
+            : (shortPayoutAmount / totalShortPriceNumber - 1) * 100,
+        )
       }
       if (longShares && collateral && !isDust(longShares, collateral.decimals)) {
-        setLongPayout((longSharesNumber || 0) * (Number(scaleBall?.value) / 100))
+        const totalLongPriceNumber = Number(formatBigNumber(totalLongPrice, collateral.decimals, collateral.decimals))
+        const longPayoutAmount = (longSharesNumber || 0) * (Number(scaleBall?.value) / 100)
+        const longProfit = longPayoutAmount - totalLongPriceNumber
+        setLongPayout(longPayoutAmount)
+        setLongProfitAmount(longProfit)
+        setLongProfitPercentage(
+          (longPayoutAmount / totalLongPriceNumber - 1) * 100 < -100
```

Comment:
Can move to const to avoid calculating twice
longProfitPercent = (longPayoutAmount / totalLongPriceNumber - 1) * 100 
setLongProfitPercentage(longProfitPercent < -100 ? -100 : longProfitPercent)